WalletWasabi/Crypto/ZeroKnowledge/NonInteractive/FiatShamirTransform/Knowledge.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections.Generic;
+using System.Linq;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge.NonInteractive.FiatShamirTransform
+{
+	public class Knowledge : IKnowledge
+	{
+		// Although in principle this could be an interface type, since only one
```

Comment:
Is it up-to-date comment?

Reply:
yes, it's meant to explain why this doesn't need to be generic (probably ever) so there's know `IStatement` or `IKnowledge` declared by the classes of the `LinearRelation` namespace, only the `IStatementDescription` interface used with the `Transcript`.